lua/nvim-treesitter/textobj.lua

```diff
@@ -0,0 +1,88 @@
+local api = vim.api
+local utils = require'nvim-treesitter.utils'
+local parsers = require'nvim-treesitter.parsers'
+local M = {}
+
+local function node_range_to_vim(node)
+  local select_range = [[
+  call cursor(%d, %d)
+  normal v
+  call cursor(%d, %d)
+  ]]
+  if node then
+    local start_row, start_col, end_row, end_col = node:range()
+
+    api.nvim_exec(string.format(select_range,
+      start_row+1, start_col+1,
+      end_row+1, end_col+1), false)
+  end
+end
+
+local function select_incremental(increment_func)
+  local buf, sel_start_line, sel_start_col, _ = unpack(vim.fn.getpos("'<"))
+  local buf, sel_end_line, sel_end_col, _ = unpack(vim.fn.getpos("'>"))
+
+  local node = nil
+  if parsers.has_parser() then
+    local root = parsers.get_parser():parse():root()
+    node = root:named_descendant_for_range(sel_start_line-1, sel_start_col-1, sel_end_line-1, sel_end_col)
+    local node_start_row, node_start_col, node_end_row, node_end_col = node:range()
+
+    if (sel_start_line-1) == node_start_row and (sel_start_col-1) == node_start_col
+      and (sel_end_line-1) == node_end_row and sel_end_col == node_end_col then
+      node = increment_func(node)
+    end
+  end
+
+  return node_range_to_vim(node)
+end
+
+function M.node_incremental()
+  -- TODO(vigoux): actually it is not really the parent here
+  -- it might be needed to climb more than only one parent, but a strictly growing node
+  return select_incremental(function(node)
+    if node then
+      return node:parent() or node
+    end
+  end)
+end
```

Comment:
you could say `M.node_incremental = select_incremental(...)`

Reply:
Yeah that's far better.